src/imitation/rewards/reward_nets.py

```diff
@@ -168,50 +71,63 @@ def _eval_reward(
             action=action,
             next_state=next_state,
             done=done,
-            device=self.device(),
-            scale=self.scale,
+            device=self.device,
+            scale=self.normalize_images,
         )
 
         with th.no_grad():
-            if is_train:
-                rew_th = self.reward_train(state_th, action_th, next_state_th, done_th)
-            else:
-                rew_th = self.reward_test(state_th, action_th, next_state_th, done_th)
+            rew_th = self(state_th, action_th, next_state_th, done_th)
 
         rew = rew_th.detach().cpu().numpy().flatten()
         assert rew.shape == state.shape[:1]
         return rew
 
+    @property
     def device(self) -> th.device:
-        """Use a heuristic to determine which device this module is on."""
-        first_param = next(self.parameters())
-        return first_param.device
-
+        """Heuristic to determine which device this module is on."""
+        try:
+            first_param = next(self.parameters())
+            return first_param.device
+        except StopIteration:
+            # if the model has no parameters, we use the CPU
+            return th.device("cpu")
 
-class RewardNetShaped(RewardNet):
-    """Abstract reward network with a phi network to shape training reward.
 
-    This RewardNet formulation matches Equation (4) in the AIRL paper.
-    Note that the experiments in Table 2 of the same paper showed shaped
-    training rewards to be inferior to an unshaped training rewards in
-    a Pendulum environment imitation learning task (and maybe HalfCheetah).
-    (See original implementation of Pendulum experiment's reward function at
-    https://github.com/justinjfu/inverse_rl/blob/master/inverse_rl/models/imitation_learning.py#L374)
-
-    To make a concrete subclass, implement `build_potential_net()` and
-    `build_base_reward_net()`.
-    """
+class ShapedRewardNet(RewardNet):
+    """A RewardNet consisting of a base net and a potential shaping."""
 
     def __init__(
         self,
         observation_space: gym.Space,
         action_space: gym.Space,
-        *,
-        discount_factor: float = 0.99,
-        **kwargs,
+        base: RewardNet,
+        potential: Callable[[th.Tensor], th.Tensor],
+        discount_factor: float,
+        normalize_images: bool = True,
     ):
-        super().__init__(observation_space, action_space, **kwargs)
-        self._discount_factor = discount_factor
+        """Setup a ShapedRewardNet instance.
+
+        Args:
+            observation_space (gym.Space): observation space of the environment
```

Comment:
as before, types in docstring not necessary when they're annotated in the function